File: shared/src/main/java/com/batu/shared/dto/request/PlaidWebhookDto.java
package com.batu.shared.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PlaidWebhookDto {

    @JsonProperty("webhook_type")
    private String webhookType;

    @JsonProperty("webhook_code")
    private String webhookCode;

    @JsonProperty("item_id")
    private String itemId;

    @JsonProperty("error")
    private PlaidWebhookErrorDto error;

    @JsonProperty("new_transactions")
    private Integer newTransactions;

    @JsonProperty("removed_transactions")
    private Integer removedTransactions;
}

File: shared/src/main/kotlin/com/batu/shared/dto/PlaidWebhookDto.kt
package com.batu.shared.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class PlaidWebhookDto(

    @JsonProperty("webhook_type")
    val webhookType: String? = null,

    @JsonProperty("webhook_code")
    val webhookCode: String? = null,

    @JsonProperty("item_id")
    val itemId: String? = null,

    @JsonProperty("error")
    val error: PlaidWebhookErrorDto? = null,

    @JsonProperty("new_transactions")
    val newTransactions: Int? = null,

    @JsonProperty("removed_transactions")
    val removedTransactions: Int? = null
)